src/main.rs

```diff
@@ -47,36 +47,88 @@ struct CalypsoArgs {
  * Reads the can socket and publishes the data to the given client.
  */
 fn read_can(pub_path: &str, can_interface: &str) -> JoinHandle<u32> {
-    //open can socket channel at name can_interface
+    // Open CAN socket channel at name can_interface
     let mut client = MqttClient::new(pub_path, "calypso-decoder");
     client.connect().expect("Could not connect to Siren!");
     let socket = CanSocket::open(can_interface).expect("Failed to open CAN socket!");
 
     thread::spawn(move || loop {
-        let msg = match socket.read_frame() {
-            Ok(CanFrame::Data(msg)) => msg,
+        // Read from MQTT socket
+        let decoded_data = match socket.read_frame() {
+            // CanDataFrame
+            Ok(CanFrame::Data(data_frame)) => {
+                let data = data_frame.data();
+                let message = DecodableMessage::new(
+                    match data_frame.id() {
+                        socketcan::Id::Standard(std) => std.as_raw().into(),
+                        socketcan::Id::Extended(ext) => ext.as_raw(),
+                    },
+                    data.to_vec(),
+                );
+                message.decode()
+            }
+            // CanRemoteFrame
             Ok(CanFrame::Remote(_)) => {
+                // TODO: Handle
+                //
```

Comment:
can you handle this by just sending the frame ID when we get a remote frame?
topic: `Calypso/Events/RemoteFrame`
unit: `id`
the value would be the ID as a float. (https://docs.rs/socketcan/latest/socketcan/frame/struct.CanRemoteFrame.html fn `id`)